### PR TITLE
Generate Jekyll documentation

### PR DESCRIPTION
**Pull request by BetsyMcPhail**
Working towards drake issue [#13832](https://github.com/RobotLocomotion/drake/issues/13832). Requires Drake PR [#14531](https://github.com/RobotLocomotion/drake/pull/14531)

### CODE REVIEW

File: driver/configurations/bazel/step-build-documentation.cmake [hunk 1]
```diff
@@ -58,6 +58,17 @@ else()
...
 
+  file(MAKE_DIRECTORY "${DASHBOARD_DOCUMENTATION_DIRECTORY}/jekyll")
+  set(BUILD_DOCUMENTATION_ARGS
+    "${DASHBOARD_BAZEL_STARTUP_OPTIONS} run ${DASHBOARD_BAZEL_BUILD_OPTIONS} //bindings/pydrake/doc:gen_jekyll -- --out_dir=${DASHBOARD_DOCUMENTATION_DIRECTORY}/jekyll")
+  separate_arguments(BUILD_DOCUMENTATION_ARGS_LIST UNIX_COMMAND "${BUILD_DOCUMENTATION_ARGS}")
+  execute_process(COMMAND ${DASHBOARD_BAZEL_COMMAND} ${BUILD_DOCUMENTATION_ARGS_LIST}
+    WORKING_DIRECTORY "${DASHBOARD_SOURCE_DIRECTORY}"
+    RESULT_VARIABLE DASHBOARD_BUILD_DOCUMENTATION_RESULT_VARIABLE)
+  if(NOT DASHBOARD_BUILD_DOCUMENTATION_RESULT_VARIABLE EQUAL 0)
+    append_step_status("BUILDING DOCUMENTATION" UNSTABLE)
```
Comment (jamiesnape) [inline driver/configurations/bazel/step-build-documentation.cmake:69]: Can you change this to `BUILDING DOCUMENTATION (JEKYLL)` and maybe disambiguate the similar messages on the rest of this page too (my fault).

**Comment by jamiesnape** [comment]
https://github.com/RobotLocomotion/drake-ci/blob/5603cd516f201bc6dfa287c7b66cf01e2ce182b8/driver/configurations/bazel/step-build.cmake#L58 is needed editing too (unfortunately).

File: driver/configurations/bazel/step-build-documentation.cmake (hunk 1 above)
Comment (BetsyMcPhail) [reply driver/configurations/bazel/step-build-documentation.cmake]: Done

**Comment by BetsyMcPhail** [comment]
> https://github.com/RobotLocomotion/drake-ci/blob/5603cd516f201bc6dfa287c7b66cf01e2ce182b8/driver/configurations/bazel/step-build.cmake#L58
> 
> is needed editing too (unfortunately).

Is ``//doc:serve_jekyll`` all that needs to be added here?

**Comment by jamiesnape** [comment]
Yes.

**Comment by BetsyMcPhail** [comment]
https://drake-jenkins.csail.mit.edu/view/Documentation/job/linux-bionic-unprovisioned-gcc-bazel-experimental-documentation/7/

**Review by jamiesnape — APPROVED** [review]

**Pull request merged**